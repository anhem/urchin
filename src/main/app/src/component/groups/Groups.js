import React, {Component} from "react";
import {Link} from "react-router-dom";

class Groups extends Component {

    render() {
        let groups = this.props.groups || [];
        let groupItems = groups.map((item, index) =>
            <li className="group-list__group" key={index.toString()}>
                <Link to={`/groups/${item.groupId}`}>{item.groupName}</Link>
            </li>
        );
        return (
            <div id="groups">
                <Link id="groups__new-group" to="/groups/new">new group</Link>
                <h2>grouplist</h2>
                <ul className="group-list">
                    {groupItems}
                </ul>
            </div>
        )
    }
}

export default Groups