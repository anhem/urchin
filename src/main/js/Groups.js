import React, {Component} from "react";

class Groups extends Component {

    render() {
        var groupList = this.props.groups.map(group =>
            <li key={group.groupId}>
                {group.name}
            </li>
        );

        return (
            <ul>
                {groupList}
            </ul>
        )
    }
}

Groups.propTypes = {
    groups: React.PropTypes.arrayOf(
        React.PropTypes.shape({
            groupId: React.PropTypes.number.isRequired,
            name: React.PropTypes.string.isRequired
        })
    )
};


export default Groups;