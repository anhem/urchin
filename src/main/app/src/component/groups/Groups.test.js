import React from 'react';
import {mount} from 'enzyme'
import {Router} from "react-router-dom";
import Groups from "./Groups";
import history from '../../history'

describe('Groups', () => {

    it('renders without crashing', () => {
        let component = mount(
            <Router history={history}>
                <Groups/>
            </Router>
        );
        expect(component.length).toEqual(1);
    });
});
