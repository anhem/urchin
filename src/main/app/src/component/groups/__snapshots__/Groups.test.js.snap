// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups match snapshot 1`] = `
<div
  id="groups"
>
  <Link
    id="groups__new-group"
    replace={false}
    to="/groups/new"
  >
    new group
  </Link>
  <h2>
    grouplist
  </h2>
  <ul
    className="group-list"
  />
</div>
`;
